lib/src/dartdoc_options.dart

```diff
@@ -597,7 +597,8 @@ class DartdocOptionArgSynth<T> extends DartdocOption<T>
       String help = '',
       OptionKind optionIs = OptionKind.other,
       this.negatable = false})
-      : assert(T is! Iterable),
+      : assert(const <Never>[] is! T,
+            '$T should not be bound to a type which allows iterables'),
```

Comment:
This would allow `T` to be `Set`, which is iterable.

To check whether `T` is/is not a subtype of `Iterable`, you need to put `T` on the LHS of the `is` check, e.g., `<T>[] is! List<Iterable>`.

Reply:
But... the comment says to not allow any type which allows an iterable. That suggests not allowing`Object` either.
In practice any non-final type may allow an iterable, so I wouldn't try to do anything for that.